tests/integration/cmd_dev_api_server_test.go

```diff
@@ -63,11 +69,157 @@ var _ = Describe("odo dev command with api server tests", func() {
 						url := fmt.Sprintf("http://%s/instance", devSession.APIServerEndpoint)
 						resp, err := http.Get(url)
 						Expect(err).ToNot(HaveOccurred())
-						// TODO: Change this once it is implemented
-						Expect(resp.StatusCode).To(BeEquivalentTo(http.StatusNotImplemented))
+						Expect(resp.StatusCode).To(BeEquivalentTo(http.StatusOK))
 					})
 				})
 			}))
 		}
+
+		When("the component is bootstrapped", helper.LabelPodmanIf(podman, func() {
+			BeforeEach(func() {
+				helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+				helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"), cmpName)
+			})
+			When("odo dev is run with --api-server flag", func() {
+				var (
+					devSession helper.DevSession
+				)
+				BeforeEach(func() {
+					opts := helper.DevSessionOpts{
+						RunOnPodman:    podman,
+						StartAPIServer: true,
+						EnvVars:        []string{"ODO_EXPERIMENTAL_MODE=true"},
+					}
+					var err error
+					devSession, err = helper.StartDevMode(opts)
+					Expect(err).ToNot(HaveOccurred())
+				})
+				AfterEach(func() {
+					devSession.Stop()
+					devSession.WaitEnd()
+				})
+				It("should serve endpoints", func() {
+					By("GETting /instance", func() {
+						url := fmt.Sprintf("http://%s/instance", devSession.APIServerEndpoint)
+						resp, err := http.Get(url)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(resp.StatusCode).To(BeEquivalentTo(http.StatusOK))
+						defer resp.Body.Close()
+						body, err := io.ReadAll(resp.Body)
+						Expect(err).ToNot(HaveOccurred())
+						strBody := string(body)
+						helper.JsonPathExist(strBody, "pid")
+						helper.JsonPathContentIs(strBody, "componentDirectory", commonVar.Context)
```

Comment:
Why this difference on macOS?

Reply:
I am not sure, it seems macOS appends `/private` to all the paths, I think it is more like a temp directory on macOS. All the directories I create with `mkdir /tmp/something`, always return `/private/tmp/something` when I `pwd` on it.

But I think this can be ignored, it might be specific to my system alone.

Some more info is available here: https://apple.stackexchange.com/a/227869